src/auth/ha-auth-external.ts

```diff
@@ -0,0 +1,69 @@
+import type { CSSResultGroup, TemplateResult } from "lit";
+import { css, html } from "lit";
+import { customElement, property } from "lit/decorators";
+import type { DataEntryFlowStepExternal } from "../data/data_entry_flow";
+import { fireEvent } from "../common/dom/fire_event";
+import type { LocalizeFunc } from "../common/translations/localize";
+import { HaForm } from "../components/ha-form/ha-form";
+
+@customElement("ha-auth-external")
+export class HaAuthExternal extends HaForm {
+  @property({ attribute: false }) public localize!: LocalizeFunc;
+
+  @property({ attribute: false }) public stepTitle?: string;
+
+  @property({ attribute: false }) public step!: DataEntryFlowStepExternal;
+
+  protected render(): TemplateResult {
+    return html`
+      <h2>${this.stepTitle}</h2>
+      <div class="content">
+        ${this.localize("ui.panel.page-authorize.external.description")}
+        <div class="open-button">
+          <a href=${this.step.url} target="_blank" rel="opener">
+            <mwc-button raised>
+              ${this.localize("ui.panel.page-authorize.external.open_site")}
+            </mwc-button>
+          </a>
+        </div>
+      </div>
+    `;
+  }
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+    const source = window.open(this.step.url);
```

Comment:
Check if `source` is not `null`, which would indicate opening was blocked by a popup blocker.

Reply:
Also rename the variable to something like `externalWindow`

Reply:
Done. Added a popup blocking error message.